include/boost/multiprecision/gmp.hpp

```diff
@@ -3734,35 +3801,39 @@ namespace Eigen
          IsSigned = std::numeric_limits<self_type>::is_specialized ? std::numeric_limits<self_type>::is_signed : true,
          RequireInitialization = 1,
       };
+
+      #if !defined(BOOST_MP_STANDALONE) || defined(BOOST_MATH_STANDALONE)
       static Real epsilon()
       {
-         return boost::math::tools::epsilon<Real>();
+         return std::numeric_limits<Real>::epsilon();
```

Comment:
This still doesn't look quite right to me: if that PP condition is met, then Boost.Math is available, and we can (and should) be using the Boost.Math functions?

Reply:
On second thoughts.... this is a traits-class specialization for gmp_float<0> only, and all the boost math functions except epsilon are defined right here in this file, and epsilon is just `ldexp(Real(1), 1 - digits())`, so the #if logic can go, and we can call the boost::math functions which are already there and do the right thing?